### PR TITLE
Add unit test of multipler orders for multiple terms aggregation.

### LINKED ISSUE #13375: [BUG] The comments did not match the codes in MultiTermsAggregationBuilder
### Describe the bug

In MultiTermsAggregationBuilder.java, the comments about multiple terms aggregation did not match codes: 
```
 GET test_000001/_search
 {
      "size": 0,
      "aggs": {
        "hot": {
          "multi_terms": {
            "terms": [{
              "field": "region"
            },{
              "field": "host"
            }],
            "order": {"max-cpu": "desc"} <====
          },
          "aggs": {
            "max-cpu": { "max": { "field": "cpu" } }
          }
        }
      }
 }
```
The comment shows we can set up one order. But from the codes, it also supports an array like:
`"order": [ {"max-cpu": "desc"} , {"max-memory": "desc"}  ]`

Here are the codes to parser this field:

`PARSER.declareObjectArray(MultiTermsAggregationBuilder::order, (p, c) -> InternalOrder.Parser.parseOrderParam(p), ORDER_FIELD);`

I also go through the codes and had some local tests. I think multiple orders do work. If so, could you please update the comment and the official document about this?

### Related component

Search:Aggregations

### To Reproduce

Compare the comments and codes.

### Expected behavior

If multiple orders are supported, we should update the comments and the official document.

### Additional Details

_No response_

### PR DESCRIPTION
**Pull request by Zhikai-VM**
### Description

I added unit tests to verify the multiple orders in the multiple terms aggregation.

### Related Issues
Resolves #13375

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml [hunk 1]
```diff
@@ -760,3 +760,75 @@ setup:
...
+        body:
+          aggs:
+            m_terms:
+              multi_terms:
+                terms:
+                  - field: str
+                  - field: boolean
+                order:
+                  - max_int: desc
+                  - min_double: asc
```
Comment (ankyhe) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml:805]: Shall we add a case for supporting size = 5?

Comment (dblock) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml:805]: Sure!

**Review by dblock — CHANGES_REQUESTED** [review]
Great stuff. 

You need to amend/sign your commits with DCO, `-s`, please?

**Comment by dblock** [comment]
Since these tests pass, make the changes in the comments as you suggested in https://github.com/opensearch-project/OpenSearch/issues/13375 and maybe open a documentation PR?

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml (hunk 1 above)
Comment (Zhikai-VM) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml:805]: Yes, I will add one more case later.

Comment (Zhikai-VM) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/370_multi_terms.yml:805]: Test case added for the 'size' of multiple terms aggregation. @ankyhe

**Comment by Zhikai-VM** [comment]
> Since these tests pass, make the changes in the comments as you suggested in #13375 and maybe open a documentation PR?

I have updated the comments in 'MultiTermsAggregationBuilder.java'.

**Comment by Zhikai-VM** [comment]
Create a doc issue for this: https://github.com/opensearch-project/documentation-website/issues/7037

**Comment by dblock** [comment]
Can you please rebase to get CI to pass? We fixed these errors in https://github.com/opensearch-project/OpenSearch/pull/13412, but need to make sure there's nothing else broken. Thanks!

**Comment by Zhikai-VM** [comment]
> Can you please rebase to get CI to pass? We fixed these errors in #13412, but need to make sure there's nothing else broken. Thanks!

Done.

**Comment by Zhikai-VM** [comment]
> Great stuff.
> 
> You need to amend/sign your commits with DCO, `-s`, please?

Done.

**Comment by Zhikai-VM** [comment]
Hi @dblock , could you please kindly let me know what should I do next? (I have signed the commits, but still shows "1 change requested")

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
> Hi @dblock , could you please kindly let me know what should I do next? (I have signed the commits, but still shows "1 change requested")

It's me! I needed to re-review and hit approve ;)

**Comment by dblock** [comment]
Thanks for contributing @Zhikai-VM! Hope you can pickup something else ;)

**Comment by Zhikai-VM** [comment]
> Thanks for contributing @Zhikai-VM! Hope you can pickup something else ;)

Thanks for your help!

**Comment by ankyhe** [comment]
@dblock We want to revise the official document also.  Could you answer this comment: https://github.com/opensearch-project/documentation-website/pull/7039#issuecomment-2089337410? Thanks.

**Comment by ankyhe** [comment]
@dblock @bowenlan-amzn Could you have a check of this PR: https://github.com/opensearch-project/documentation-website/pull/7039.  Thanks.

**Pull request merged**